src/xo-app/tasks/index.js

```diff
@@ -140,12 +144,30 @@ const GROUPED_ACTIONS = [
 })
 @injectIntl
 export default class Tasks extends Component {
-  _showPoolTasks = pool =>
-    isEmpty(this.state.pools) ||
-    includes(resolveIds(this.state.pools), resolveId(pool))
+  componentWillMount () {
+    this._autoFill(this.props)
+  }
+
+  componentWillReceiveProps (props) {
+    this._autoFill(props)
+  }
+
+  _autoFill = props => {
```

Comment:
Why do you need this?

Reply:
I change the state of pools as soon as props are changed, and I use it to take tasks by pool.

Reply:
The problem with this (apart from the fact that this will run every time the props change) is that if I select a pool and then unselect it, it will filter out *all* the pools and the SortedTable will be empty until the props change again. The `_showPoolTasks` was actually a good solution for this even though it could be improved via a selector if we consider the `includes`/`resolveIds` is a heavy computation.
So what about something like this?

```js
_getTasks = createSelector(
  createSelector(() => this.state.pools, resolveIds),
  () => this.props.pendingTasksByPool,
  (poolIds, pendingTasksByPool) => isEmpty(poolIds)
    ? flatten(toArray(pendingTasksByPool))
    : flatMap(pools, pool => pendingTasksByPool[pool.id])
)
```
Then you wouldn't even have to call `_showPoolTasks` in the return.